File: cpp/118_Pascals_Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ret;

        for (int i = 0; i < numRows; ++i) {
            vector<int> row(i + 1, 1);
            ret.push_back(row);
        }

        for (int i = 2; i < numRows; ++i) {
            for (int j = 1; j < i; ++j) {
                ret[i][j] = ret[i - 1][j - 1] + ret[i - 1][j];
            }
        }

        return ret;
    }
};

File: py/118_Pascals_Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ret = [] 
        for i in range(numRows):
            _ = [1] * (i + 1)
            ret.append(_)

        for i in range(2, numRows):
            for j in range(1, i):
                ret[i][j] = ret[i - 1][j - 1] + ret[i - 1][j]

        return ret
